java/src/main/java/ai/rapids/cudf/CuFile.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ai.rapids.cudf;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+
+/**
+ * JNI wrapper for accessing the cuFile API.
+ */
+public class CuFile {
+  private static final Logger log = LoggerFactory.getLogger(CuFile.class);
+  private static boolean initialized = false;
+  private static long driverPointer = 0;
+
+  static {
+    initialize();
+  }
+
+  /**
+   * Load the native libraries needed for libcufilejni, if not loaded already; open the cuFile
+   * driver, and add a shutdown hook to close it.
+   */
+  private static synchronized void initialize() {
+    if (!initialized) {
+      try {
+        NativeDepsLoader.loadNativeDeps(new String[]{"cufilejni"});
+        driverPointer = createDriver();
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+          destroyDriver(driverPointer);
+        }));
+        initialized = true;
+      } catch (Throwable t) {
+        log.error("Could not load cuFile jni library...", t);
+      }
+    }
+  }
+
+  private static native long createDriver();
+
+  private static native void destroyDriver(long pointer);
+
+  /**
+   * Copy a device buffer to a given file path.
+   *
+   * @param path The file path to copy to.
+   * @param buffer The device buffer to copy from.
+   * @param append Whether to append to the file.
+   * @return The file offset from which the buffer was appended.
+   */
+  public static long copyDeviceBufferToFile(File path, DeviceMemoryBuffer buffer, boolean append) {
```

Comment:
nit: I know this is the first go at this API, but I would love to see it expose the same info that `cuFileRead` and `cuFileWrite` do so we don't have to update APIs later on if we need more functionality.

https://docs.nvidia.com/gpudirect-storage/api-reference-guide/index.html#cufile-io-api

In this case I think it would be nice to have an API that took a start offset and length for buffer, and an offset in path, perhaps as an alternative to append.

Reply:
For the buffer offset, would it be better to pass in a `DeviceMemoryBufferView`?

For the file offset, when we are appending, typically we don't know the actual offset.

Reply:
I think I would rather just have a long offset value, assuming that we check the bounds for the copy too, but if you think a `DeviceMemoryBufferView` would be cleaner we probably could just set the type to `BaseDeviceMemoryBuffer` which will cover either class.

Reply:
For the file offset, I realize that we don't know the offset. I was hoping for two APIs one that does an append and another that would take an offset and allow the user to give an offset.

Reply:
Done.